File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.java
class Solution {
    public int findMin(int[] nums) {
        int min = 1000;
        int l = 0;
        int r = nums.length - 1;

        while(l<=r){
            int m = l + (r - l) / 2;
            if(nums[m] > nums[r]){
                l = m + 1;
            } else{
                r = m - 1;
            }

            if(nums[m] < min){
                min = nums[m];
            }
        }

        return min;
    }
}

File: Data Structures & Algorithms/jump-game-ii/submission-0.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        farthest = math.inf
        r = len(nums) - 1
        total = 0
        while r > 0:
            for i in range (r-1,-1,-1):
                if nums[i] + i >= r:
                    farthest = min(farthest,i)

            r = farthest
            total+=1

        return total


            
